### DISCUSSION TITLE
Adding link to docker image from GHCR to Release

### DISCUSSION THREAD

**Discussion by malovicg** [community/community#122439] [Actions]
### Select Topic Area

General

### Body

Hi,

I am new to GitHUb actions, and I am trying to make a workflow for publishing docker image. These are the steps involved:

- build / test the app
- Maintain the app version using [Versionize](https://github.com/versionize/versionize)
- Create docker image and push it to GitHub container registry (GHCR)
- Release (this is were I am at now)
- Deploy

Below is workflow I have now. I was able to create a Release that contains hangelog and source code. I would like to be able to have a link to docker image In Release as well.

So, I need to:
1) get a link to the docker image I pushed to GHCR
2) add link to Release

**Is this possible to achieve?**

Thanks,
Goran

`name: Build, test, release and deploy docker Image to GHCR

on:
  push:
    branches: [ "master" ]

jobs:

  build:

    runs-on: ubuntu-latest

    steps:
    - name: Checkout GitHub Action
      uses: actions/checkout@v4
      with:
        fetch-depth: 0

    - name: Build
      run: dotnet build
        
    - name: Install Versionize
      run: dotnet tool install --global Versionize
    - name: Setup git
      run: |
        git config --local user.email "my@email.com"
        git config --local user.name "My Name"
    - name: Versionize Release
      id: versionize
      run: versionize --changelog-all --skip-dirty
      continue-on-error: false
    - name: No release required
      if: steps.versionize.outcome != 'success'
      run: echo "Skipping Release. No release required."     
    - name: Push changes to GitHub
      if: steps.versionize.outcome == 'success'
      uses: ad-m/github-push-action@master
      with:
        github_token: ${{ secrets.GITHUB_TOKEN }}
        branch: ${{ github.ref }}
        tags: true
        
    - name: Get version
      if: steps.versionize.outcome == 'success'
      uses: "actions/github-script@v7.0.1"
      id: get-version
      with:
        script: |
          try {
            const tags_url = context.payload.repository.tags_url + "?per_page=1"
            const result = await github.request(tags_url)
            const current_tag = result.data[0].name
            return current_tag
          } catch (error) {
            core.setFailed(error.message);
          }

    - name: Login to GitHub Container Registry
      uses: docker/login-action@v3
      with:
        registry: ghcr.io
        username: ${{github.actor}} 
        password: ${{secrets.GITHUB_TOKEN}}
    - name: Build the Docker image
      run: |
        repository=ghcr.io/${GITHUB_REPOSITORY}
        image=${repository}/bm-web
        docker build --file Web/Web/Dockerfile --tag ${image}:latest --tag ${image}:${{steps.get-version.outputs.result}} --tag ${image}::$(date +%s)} .
        docker push ${image} --all-tags
        
    - name: Create release
      if: steps.versionize.outcome == 'success'
      uses: "actions/github-script@v7.0.1"
      with:
        github-token: "${{ secrets.GITHUB_TOKEN }}"
        script: |
          try {
            await github.rest.repos.createRelease({
              draft: false,
              generate_release_notes: true,
              name: ${{steps.get-version.outputs.result}},
              owner: context.repo.owner,
              prerelease: false,
              repo: context.repo.repo,
              tag_name: ${{steps.get-version.outputs.result}},
              changelog : {
                title: "Changelog",
                body: "This is the changelog"
              }
            });
          } catch (error) {
            core.setFailed(error.message);
          }`

**Comment by ivan-penchev** [comment]
Let to the party but...
the solution to this is, you need to fetch your latest docker pusher container from the ghcr with that tag, then extract the version id for that tag, and compose the url
This is how I did it
```yaml
      - name: Fetch package version ID
        id: fetch_version_id
        run: |
          new_tag=${{ needs.fetch_previous_calculate_next_tags.outputs.new_tag }}
          package_name=${{ env.IMAGE_NAME }}
          repo_name=${{ github.repository }}
          owner=${{ github.repository_owner }}
          version_id=$(curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
              "https://api.github.com/orgs/${owner}/packages/container/${package_name}/versions" | \
              jq -r --arg tag "$new_tag" '.[] | select(.metadata.container.tags[] == $tag) | .id')
          package_url="https://github.com/${repo_name}/pkgs/container/${package_name}/${version_id}?tag=${new_tag}"
          echo "package_url=$package_url" >> $GITHUB_OUTPUT
      - name: Write Docker image build summary
        run: |
          echo "### Docker Image Build Summary :whale:"  >> $GITHUB_STEP_SUMMARY
          echo "Docker image built and pushed to [${{ env.REGISTRY }}/${{ github.repository_owner }}/${{ env.IMAGE_NAME }}:${{ needs.fetch_previous_calculate_next_tags.outputs.new_tag }}](${{ steps.fetch_version_id.outputs.package_url }})" >> $GITHUB_STEP_SUMMARY
```
